ultralytics/engine/exporter.py

```diff
@@ -702,15 +714,109 @@ def export_engine(self, prefix=colorstr("TensorRT:")):
                 LOGGER.warning(f"{prefix} WARNING ⚠️ 'dynamic=True' model requires max batch size, i.e. 'batch=16'")
             profile = builder.create_optimization_profile()
             min_shape = (1, shape[1], 32, 32)  # minimum input shape
-            opt_shape = (max(1, shape[0] // 2), *shape[1:])  # optimal input shape
+            opt_shape = tuple(shape)  # optimal input shape
             max_shape = (*shape[:2], *(max(1, self.args.workspace) * d for d in shape[2:]))  # max input shape
             for inp in inputs:
                 profile.set_shape(inp.name, min_shape, opt_shape, max_shape)
```

Comment:
`opt_shape` could be removed and replaced with `shape`, with L717 no longer required